### DISCUSSION TITLE
Alternative genetic crossover scheme inspired by biology

### DISCUSSION THREAD

**Discussion by colgreen** [colgreen/sharpneat#59] [Ideas]
Biological sexual reproduction works like so (high level):

 * Both mother and father have two sets of DNA each. I.e. each cell contain two copies, one from their mother (egg), and one from their father (sperm)
 * When an egg is formed, the two sets of DNA are merged/collapsed in some way to produce one set of DNA, i.e. crossover occurs wen the egg is formed.
 * The same or similar happens to form a sperm.
 * The egg and the sperm come together to form two sets of DNA, these are now carried by the new embyro/child in each of their cells.

This is quite different to how the canonical genetic algorithm operates, i.e. with a single 'strand' of genetic material in each individual. The biological approach is likely more powerful, as it produces more variation in the offspring, although you do have to apply rules to convert two strands of DNA into a functional phenome - so what would those rules be for something like NEAT?

**Comment by tlemo** [comment]
Sounds like a diploid vs. haploid genotype, right? It would be interesting to experiment with a diploid NEAT version, perhaps using a scheme of dominance/codominance for gene expression - although NEAT direct encoding seems to present a challenge. It would be exciting to see experiments in this area.

> When an egg is formed, the two sets of DNA are merged/collapsed in some way to produce one set of DNA, i.e. crossover occurs wen the egg is formed.

My understanding is that crossover technically happens during meiosis (splitting a diploid cells into haploid daughter cells), have I read this wrong?

**Comment by colgreen** [comment]
> Sounds like a diploid vs. haploid genotype, right?

Yes, I think so.

> It would be interesting to experiment with a diploid NEAT version, perhaps using a scheme of dominance/codominance for gene expression - although NEAT direct encoding seems to present a challenge. It would be exciting to see experiments in this area.

Yeah, I don't have clear ideas on how this would actually work. Any thoughts/ideas are welcome!


> > When an egg is formed, the two sets of DNA are merged/collapsed in some way to produce one set of DNA, i.e. crossover occurs wen the egg is formed.
> 
> My understanding is that crossover technically happens during meiosis (splitting a diploid cells into haploid daughter cells), have I read this wrong?

I'm sure you're right. I should brush up and read https://en.wikipedia.org/wiki/Meiosis before I work on this task!
